mongo/options/clientoptions.go

```diff
@@ -1153,38 +1013,26 @@ func (c *ClientOptionsBuilder) SetWriteConcern(wc *writeconcern.WriteConcern) *C
 // compressor through ApplyURI or SetCompressors. Supported values are -1 through 9, inclusive. -1 tells the zlib
 // library to use its default, 0 means no compression, 1 means best speed, and 9 means best compression.
 // This can also be set through the "zlibCompressionLevel" URI option (e.g. "zlibCompressionLevel=-1"). Defaults to -1.
-func (c *ClientOptionsBuilder) SetZlibLevel(level int) *ClientOptionsBuilder {
-	c.Opts = append(c.Opts, func(opts *ClientOptions) error {
-		opts.ZlibLevel = &level
-
-		return nil
-	})
+func (c *ClientOptions) SetZlibLevel(level int) *ClientOptions {
+	c.ZlibLevel = &level
 
 	return c
 }
 
 // SetZstdLevel sets the level for the zstd compressor. This option is ignored if zstd is not specified as a compressor
 // through ApplyURI or SetCompressors. Supported values are 1 through 20, inclusive. 1 means best speed and 20 means
 // best compression. This can also be set through the "zstdCompressionLevel" URI option. Defaults to 6.
-func (c *ClientOptionsBuilder) SetZstdLevel(level int) *ClientOptionsBuilder {
-	c.Opts = append(c.Opts, func(opts *ClientOptions) error {
-		opts.ZstdLevel = &level
-
-		return nil
-	})
+func (c *ClientOptions) SetZstdLevel(level int) *ClientOptions {
+	c.ZstdLevel = &level
 
 	return c
 }
 
 // SetAutoEncryptionOptions specifies an AutoEncryptionOptions instance to automatically encrypt and decrypt commands
 // and their results. See the options.AutoEncryptionOptions documentation for more information about the supported
 // options.
-func (c *ClientOptionsBuilder) SetAutoEncryptionOptions(aeopts Lister[AutoEncryptionOptions]) *ClientOptionsBuilder {
-	c.Opts = append(c.Opts, func(opts *ClientOptions) error {
-		opts.AutoEncryptionOptions = aeopts
-
-		return nil
-	})
+func (c *ClientOptions) SetAutoEncryptionOptions(aeopts Lister[AutoEncryptionOptions]) *ClientOptions {
```

Comment:
For `AutoEncryptionOptions`, `LoggerOptions`, and `ServerAPIOptions`, should we also revert to the previous pattern (i.e. not use of `Lister` type)?